solanaj/src/main/java/org/p2p/solanaj/utils/PublicKeyValidator.kt

```diff
@@ -0,0 +1,22 @@
+package org.p2p.solanaj.utils
+
+import org.bitcoinj.core.Base58
+import org.p2p.solanaj.core.PublicKey
+
+object PublicKeyValidator {
+    private const val REGEX = "[1-9A-HJ-NP-Za-km-z]{32,44}"
+
+    @Throws(IllegalArgumentException::class)
+    fun validate(publicKey: String) = require(
+        Regex(REGEX).matches(publicKey) && Base58.decode(publicKey).size == PublicKey.PUBLIC_KEY_LENGTH
```

Comment:
you can use Base58Utils here.

Reply:
What's the profit of `encode()`, `decode()` functions from `Base58Utils`? It's like just an alias for `Base58` corresponding functions

Reply:
agreed, but the main thing here is that Base58Utils is just ONE point of entry for these methods, so if we wanted to add logging to decode/encode actions, it would be much easier to do so in one place. 
also I think it would be easier to mock later

Reply:
I see, okay